pkg/shell/base_index.js

```diff
@@ -140,14 +147,31 @@ var phantom_checkpoint = phantom_checkpoint || function () { };
                 frame.style.display = "none";
 
                 var base, checksum;
-                if (machine)
-                    checksum = machine.checksum;
-                if (host === "localhost")
-                    base = "..";
-                else if (checksum)
-                    base = "../../" + checksum;
-                else
+                if (machine) {
+                    if (machine.manifests && machine.manifests[".checksum"])
+                        checksum = "$" + machine.manifests[".checksum"];
```

Comment:
And to elaborate why we need it also in the manifest: We always need the machine checksum now in order to decide whether we can use the package checksum in the URL.  Previously, we would get the machine checksum when retrieving the manifest, but the manifest is not retrieved when we find a cached version in `window.sessionStorage` (I think).